softgroup/services/profile/profile-impl/src/main/java/com/softgroup/profile/impl/handler/SetMyProfileHandler.java

```diff
@@ -33,18 +39,27 @@ public String getName() {
         SetMyProfileRequest requestData = request.getData();
         SetMyProfileResponse setMyProfileResponse = new SetMyProfileResponse();
 
-        profileService.insertProfile(requestData.getProfileEntities());
-
-
+        ProfileDTO profileDTO = requestData.getProfile();
         Response<SetMyProfileResponse> response = new Response<SetMyProfileResponse>();
         response.setHeader(request.getHeader());
         response.setData(setMyProfileResponse);
-
         ResponseStatus status = new ResponseStatus();
-        status.setCode(200);
-        status.setMessage("OK");
 
-        response.setStatus(status);
-        return null;
+        try {
+            ProfileEntity profileEntity = (ProfileEntity) mapper.map(profileDTO, ProfileEntity.class);
+            profileEntity.setId(request.getRoutingData().getProfileId());
+            profileService.insertProfile(profileEntity);
```

Comment:
@VadymPolyanski You are trying to recreate profile entity. But must only update certain fields